src/include/distributed/version_compat.h

```diff
@@ -44,6 +47,102 @@ PathNameOpenFilePerm(FileName fileName, int fileFlags, int fileMode)
 }
 
 
+static inline MemoryContext
+AllocSetContextCreateExtended(MemoryContext parent,
+							  const char *name,
+							  Size minContextSize,
+							  Size initBlockSize,
+							  Size maxBlockSize)
+{
+	return AllocSetContextCreate(parent, name, minContextSize, initBlockSize,
+								 maxBlockSize);
+}
+
+
+static inline void
+ExplainPropertyIntegerExtended(const char *qlabel, const char *unit,
+							   int64 value, ExplainState *es)
+{
+	return ExplainPropertyInteger(qlabel, value, es);
+}
+
+
+static inline List *
+ExtractVacuumTargetRels(VacuumStmt *vacuumStmt)
+{
+	List *vacuumList = NIL;
+
+	if (vacuumStmt->relation != NULL)
+	{
+		vacuumList = list_make1(vacuumStmt->relation);
+	}
+
+	return vacuumList;
+}
+
+
+static inline List *
+VacuumColumnList(VacuumStmt *vacuumStmt, int relationIndex)
+{
+	Assert(relationIndex == 0);
+	return vacuumStmt->va_cols;
+}
+
+
+#define DatumGetJsonbP(d) DatumGetJsonb(d)
+
+
+#endif
+
+
+#if (PG_VERSION_NUM >= 110000)
+#include "optimizer/prep.h"
+
+static inline void
+ExplainPropertyIntegerExtended(const char *qlabel, const char *unit,
+							   int64 value, ExplainState *es)
+{
+	return ExplainPropertyInteger(qlabel, unit, value, es);
+}
+
+
+static inline Expr *
+canonicalize_qual_ext(Expr *qual, bool is_check)
+{
+	return canonicalize_qual(qual, is_check);
+}
+
+
+static inline List *
+ExtractVacuumTargetRels(VacuumStmt *vacuumStmt)
+{
+	List *vacuumList = NIL;
+
+	ListCell *vacuumRelationCell = NULL;
+	foreach(vacuumRelationCell, vacuumStmt->rels)
+	{
+		VacuumRelation *vacuumRelation = (VacuumRelation *) lfirst(vacuumRelationCell);
+		vacuumList = lappend(vacuumList, vacuumRelation->relation);
+	}
+
+	return vacuumList;
+}
+
+
+static inline List *
+VacuumColumnList(VacuumStmt *vacuumStmt, int relationIndex)
+{
+	VacuumRelation *vacuumRelation = NULL;
+
+	Assert(relationIndex >= 0 && list_length(vacuumStmt->rels) > relationIndex);
```

Comment:
I could be wrong but I thought many of the list utilities already have asserts similar to this?